Update [Feature:PetSet] e2e tests to [Feature:StatefulSet]

janetkuo opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://github.com/kubernetes/kubernetes/blob/master/CONTRIBUTING.md and developer guide https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md
2. If you want *faster* PR reviews, read how: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/faster_reviews.md
3. Follow the instructions for writing a release note: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/pull-requests.md#release-notes
-->

**What this PR does / why we need it**: Update StatefulSet e2e test feature tags from [Feature:PetSet] to [Feature:StatefulSet]

**Special notes for your reviewer**: Ref test-infra changes https://github.com/kubernetes/test-infra/pull/1151

cc @erictune @foxish @kow3ns @enisoc @kubernetes/sig-apps

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access) 
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`. 
-->
```release-note
NONE
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37243)
<!-- Reviewable:end -->

janetkuo committed:
Update [Feature:PetSet] e2e tests to [Feature:StatefulSet]

foxish commented:
This change would need to go in with a corresponding change in test-infra to test the [StatefulSet] jobs. And while we're at it, we should also stop the alpha Jenkins job from testing StatefulSets.

foxish commented:
Oh, just noticed the other PR. Thanks!

janetkuo commented:
@k8s-bot bazel test this

janetkuo commented:
Maybe we should just remove all [Feature] tag from StatefulSet tests, since it's not alpha anymore?

mikedanese commented:
@k8s-bot bazel test this

janetkuo commented:
Per discussion with @bprashanth, updated all Stateful Set e2e tests:

1. Network partition: Removed `Feature` tag, keep `Disruptive` tag, they now runs in `Serial` tests (instead of PetSet feature tests)
2. Clustered apps (MySQL, zookeeper, etc): Updated `[Feature:PetSet]` to `[Feature:StatefulSet]`, it now runs in StatefulSet feature tests (instead of PetSet)
3. Other StatefulSet e2e tests: Removed `Feature` tag, keep `Slow` tag, they now runs in `Slow` tests (instead of PetSet feature tests)

foxish commented on test/e2e/petset.go:
```diff
@@ -141,7 +141,7 @@ var _ = framework.KubeDescribe("StatefulSet [Slow] [Feature:PetSet]", func() {
 		})
 
-		It("should handle healthy pet restarts during scale [Feature:PetSet]", func() {
+		It("should handle healthy pet restarts during scale", func() {
```
References "pet", but I can see other places in this file where that is the case as well. Do we want to change the wording in a follow-up later on?

janetkuo replied:
Yeah I can change this in this PR, in a separate commit

janetkuo replied:
Changed my mind, this will be a follow-up PR

janetkuo replied:
See https://github.com/kubernetes/kubernetes/pull/37330

foxish commented on test/e2e/petset.go:
```diff
@@ -67,7 +67,7 @@ const (
 // Time: 25m, slow by design.
```
Which of these tests are actually slow? I'm not sure that they take more than 5 mins to run on average.

janetkuo replied:
I checked the logs on GCE, all stateful set tests (other than disruptive and clustered app ones) took <200s. In this case I'll remove `Slow` tag on them.

janetkuo committed:
Remove [Slow] from statefulset basic tests

foxish commented:
/lgtm
